<!--Designed by Shrey Gupta from the Translational Genomics Research Institute in 2015 for Dr. Seungchan Kim. The following code was made using a template created by Shrey Gupta and protected under the GNU GPL v.2.0 License.-->

<!--TODO: 
	1) Add in Zero Clipboard for copying of code http://zeroclipboard.org/
	2) Add the link to the previous version of EDDY.
-->

<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
	<title>EDDY: Evaluation of Differential DependencY</title>
	<script src="./assets/jquery/jquery-1.x-min.js"></script>
	<script src="./assets/bs3css/js/bs3.min.js"></script>
	<link rel="stylesheet" href="./assets/bs3css/css/bs3.min.css" />
	<script src="./assets/scripts/common.js"></script>
	<link rel="stylesheet" href="./assets/styles/common.css" />
</head>
<body>
	<div class="container-fluid">
		<div class="row">
			<nav class="col-xs-12 navbar navbar-default eddy-navbar">
				<div class="container-fluid">
					<div class="navbar-header">
						<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
							<span class="sr-only">Toggle navigation</span>
							<span class="icon-bar"></span>
							<span class="icon-bar"></span>
							<span class="icon-bar"></span>
						</button>
						<a class="navbar-brand" href="Home.html">EDDY</a>
					</div>
					<div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
						<ul class="nav navbar-nav">
							<li class="active"><a href="EDDY_old.html">OLD EDDY<span class="sr-only">(current)</span></a></li>
							<li><a href="#">Some Link</a></li>
							<li class="dropdown">
								<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Some Dropdown<span class="caret"></span></a>
								<ul class="dropdown-menu" role="menu">
									<li><a href="#">Some Link</a></li>
									<li><a href="#">Some Link</a></li>
									<li><a href="#">Some Link</a></li>
									<li class="divider"></li>
									<li><a href="#">Some Link</a></li>
									<li class="divider"></li>
									<li><a href="#">Some Link</a></li>
								</ul>
							</li>
						</ul>
						<ul class="nav navbar-nav navbar-right">
							<li><a href="#">Some Link</a></li>
							<li class="dropdown">
								<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Some Dropdown<span class="caret"></span></a>
								<ul class="dropdown-menu" role="menu">
									<li><a href="#">Some Link</a></li>
									<li><a href="#">Some Link</a></li>
									<li><a href="#">Some Link</a></li>
									<li class="divider"></li>
									<li><a href="#">Some Link</a></li>
								</ul>
							</li>
						</ul>
					</div>
				</div>
			</nav>			
		</div>
		<div class="row">
			<div class="jumbotron eddy-banner col-xs-12">
				<h1>EDDY: Evaluation of Differential DependencY</h1>
				<p class="eddy-page-description">This page provides an updated Java implementation of an algorithm EDDY. The original version, which was presented in "<a href="http://nar.oxfordjournals.org/content/42/7/e60">EDDY: A novel statistical gene set test method to detect differential genetic dependencies</a>", can be found <a href="EDDY_old.html">here</a>.</p>
			</div>
		</div>
		<div class="row">
			<div class="container eddy-content">
				<div class="row">
					<div class="col-xs-12 col-md-9 eddy-mainbar">
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title sp-eddy-mainbar-section-title-first"><a name="what-is-eddy">What is EDDY?</a></h1>
							<p>Identifying differential features between conditions is a popular approach to understanding molecular features and their mechanisms underlying a biological process of particular interest. Although many tests for identifying differential expression of gene or gene sets have been proposed, there was limited success in developing methods for differential interactions of genes between conditions because of its computational complexity. We present a method for Evaluation of Dependency DifferentialitY (EDDY), which is a statistical test for differential dependencies of a set of genes between two conditions. Unlike previous methods focused on differential expression of individual genes or correlation changes of individual gene-gene interactions, EDDY compares two conditions by evaluating the probability distributions of dependency networks from genes. The method has been evaluated and compared with other methods through simulation studies, and application to glioblastoma multiforme data resulted in informative cancer and glioblastoma multiforme subtype-related findings. The comparison with Gene Set Enrichment Analysis, a differential expression-based method, revealed that EDDY identifies the gene sets that are complementary to those identified by Gene Set Enrichment Analysis. EDDY also showed much lower false positives than Gene Set Co-expression Analysis, a method based on correlation changes of individual gene-gene interactions, thus providing more informative results. The Java implementation of the algorithm is freely available to noncommercial users.</p>
						</div>
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title"><a name="downloads">Downloads</a></h1>
							<div class="row">
								<div class="col-xs-12">
									<h3>Command Line EDDY</h3>
									<p><code>eddy_mt-v2_5_2.jar</code> is a runnable JAR file that can be used to run a single instance of EDDY from the command line. This can now be run over multiple gene sets. </p>
									<a class="btn btn-primary eddy-download-link" href="./download/v2_5_2/eddy_mt-v2_5_2.jar"><span class="glyphicon glyphicon-download-alt"></span> Download eddy_mt-v2_5_2.jar</a>
								</div>
							</div>
						</div>
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title"><a name="system-requirements">System Requirements</a></h1>
							<p>Programs were implemented and tested on Java version 1.6.0. Requires Java virtual machine 1.6.0 or higher. EDDY is a multi-threaded application and has been tested up to 32 cores.</p>
						</div>
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title"><a name="file-formats">File Formats</a></h1>
							<div class="row"><div class="col-xs-12">
								<h3>Input data file</h3><p>The input data file is a tab-delimited text file. Each row is a variable and each column is a sample. It needs both of row and column headers for variable names and sample names. Variable values should be categorial. The following file is a sample input data file:</p>
								<div class="row eddy-download-module-small">
									<div class="col-sm-5"><a class="btn btn-primary eddy-download-link-small eddy-download-link" href="./download/v2_5_2/TCGA_GBM_mRNA_discrete_HGNC.txt"><span class="glyphicon glyphicon-download-alt"></span> TCGA_GBM_mRNA_discrete_HGNC.txt</a></div>
									<div class="col-sm-7 eddy-download-description">Glioblastoma gene expression data from The Cancer Genome Atlas (TCGA). Refer the publication for preprocessing information.</div>
								</div>
							</div></div>
							<div class="row"><div class="col-xs-12">
								<h3>Sample class information file</h3><p>The sample class information file is a tab-delimited single line text file. Two-class information needs to be listed in the order of samples from the input data file. The following file is an example of a sample class information file:</p>
								<div class="row eddy-download-module-small">
									<div class="col-sm-3"><a class="btn btn-primary eddy-download-link-small eddy-download-link" href="./download/v2_5_2/Proneural.txt"><span class="glyphicon glyphicon-download-alt"></span> Proneural.txt</a></div>
									<div class="col-sm-9 eddy-download-description">Sample class information with Proneural subtype versus the other subtypes.</div>
								</div>
							</div></div>
							<div class="row"><div class="col-xs-12">
								<h3>Gene set file (for test)</h3><p>The gene set file is a tab-delimited mutiple line text file where each line provides the name of the gene set, source URL and gene names (or variable names for general applications). The following file is a sample combined gene set file:</p>
								<div class="row eddy-download-module-list">
									<div class="col-sm-12"><a class="btn btn-primary eddy-download-link-small eddy-download-link" href="./download/v2_5_2/c2.cp.v4.0.symbols.gmt.txt"><span class="glyphicon glyphicon-download-alt"></span> c2.cp.v4.0.symbols.gmt.txt</a></div>
								</div>
							</div></div>
						</div>
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title"><a name="operating-the-cli">Operating the Command Line EDDY Program</a></h1>
							<p>For command line options, execute the following command:</p>
							<div class="well">
								<code>% java -jar eddy_mt-v2_5_2.jar</code>
							</div>
							<p>You may use the following options with the command:</p>
							<table class="eddy-table-o-variables">
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-d</code></td>
									<td>input data file</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-g</code></td>
									<td>gene set list file</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-c</code></td>
									<td>class information file</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-edf</code></td>
									<td>edge-direction fixed boolean parameter</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-m</code></td>
									<td>minimum gene set size for analysis. Default = -1 for no limit.</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-M</code></td>
									<td>maximum gene set size for analysis. Default = -1 for no limit.</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-mp</code></td>
									<td>maximum number of parents for each node</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-mt</code></td>
									<td>number of threads to use</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-p</code></td>
									<td>pvalue threshold for independence testing. [default = 0.05]</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-qp</code></td>
									<td>Quick permutations enabled -- permutation stopswhen there is no possibility to yield p-value less than given threshold.</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-s</code></td>
									<td>number network structures to consider [default = LOO (-1)]</td>
								</tr>
								<tr>
									<td class="sp-eddy-cli-variable-names"><code>-r</code></td>
									<td>number of permutations for statistical significance testing. [default = 1000]</td>
								</tr>
							</table>
							<p>The following is an example command:</p>
							<div class="well">
								<code>% java -jar eddy_mt-v2_5_2.jar -d TCGA_GBM_mRNA_discrete_HGNC.txt -g c2.cp.v4.0.symbols.gmt.txt -c Proneural.txt -m 5 -M 50 -r 1000</code>
							</div>
							<p>In case of low memory error (<code>java.lang.OutOfMemoryError Exception</code>, from Java heap space), use <code>-Xmx</code> option for Java virtual machine to increase the amount of memory. For example:</p>
							<div class="well">
								<code>% java -Xmx8196m -jar eddy_mt-v2_5_2.jar</code><br/>
								(Use up to 8196MB heap space for this execution of <code>eddy_mt-v2_5_2.jar)</code>.)
							</div>
							<p>Based on the specified size limits (<code>-m</code> and <code>-M</code> options), EDDY constructs a list of gene sets and sorts it from largest to smallest. An EDDY process is generated for each gene set and distributed to the next available thread in round robin fashion.</p>
						</div>
						<div class="eddy-mainbar-section">
							<h1 class="eddy-mainbar-section-title"><a name="post-processing-eddy-output">Post Processing EDDY Output</a></h1>
							<p>For gene sets that meet the p-value criterion, EDDY generates four files:</p>
							<ol>
								<li>A <code>Network.txt</code> file specifying all the network configurations that were analyzed</li>
								<li>A <code>summary.txt</code> file, listing the probabilities of these networks</li>
								<li>An adjacency matrix representing a composite of the most likely network configurations</li>
								<li>An <code>Edgelist.txt</code> file for this same composite network</li>
							</ol>
							<p>For display purposes, an additional <code>Nodes.txt</code> file can be generated, which gives the ratio of interaction types on incoming edges for each node. A python script to create this is provided <a href="./download/v2_5_2/edge_counter.py">here</a>. It is invoked as:</p>
						</div>
						<div class="well">
							<code>% python edge_count.py PATHWAYNAME_Edgelist.txt</code><br/>
						</div>
						<p>To display a network configuration for a specific geneset using <a href="http://www.cytoscape.org/">Cytoscape 3.2</a>.</p>
						<ol>
							<li>Choose "Network from File" and select the <code>Edgelist.txt</code> for the specific geneset. Choose the interaction from column 3 and source/destination columns 1 &amp; 2. To avoid skipping the first line, it will work if you check the "Show text file import options" box and then uncheck "Space". The top line of the file will now appear in the edge list.</li>
							<li>Go to import <kbd>File</kbd> &#8594; <kbd>Import</kbd> &#8594; <kbd>Table</kbd> &#8594; <kbd>File</kbd> and import the "Nodes" file. Then click "OK."</li>
							<li>Under <kbd>Style</kbd> &#8594; <kbd>Edge</kbd> &#8594; <kbd>Stroke Color</kbd>, set Column to "Interaction" and Mapping Type to "Discrete Mapping" and assign a color to each interaction by clicking on the right edge of each cell.</li>
							<li>Now, under <kbd>Style</kbd> &#8594; <kbd>Node</kbd> &#8594; <kbd>Border Paint</kbd>, set Column to one of the ratio columns and Mapping Type to "Continuous Mapping” and click on the color map.  Clicking on the small triangles ("handles") on the top sets the colors.  Keep these consistent with the edge colors you picked.  Slide left triangle all the way left.  Move the second one to the center and assign a neutral (e.g. grey) color.  <kbd>Add</kbd> another triangle and shift it all the way right.</li>
						</ol>
						<p>This should create a network image like pictured below.</p>
					<img src="assets/imgs/sample_graph.png">
					<div class="eddy-mainbar-section">
					<h1 class="eddy-mainbar-section-title"><a name="Acknowledgements">Acknowledgements</a></h1>
					<p>The development of the software EDDY was partly funded by National Cancer Institute, National Institutes of Health (NIH) [29XS195 — In-Silico Research Center of Excellence (<a href="http://cabig.cancer.gov/action/collaborations/isrc/">ISRCE</a>), 1U01CA168397-01 — Center for Target Discovery and Development (<a href="https://ocg.cancer.gov/programs/ctd2">CTD2</a>) / Targeted Treatment for Tumors (<a href="https://www.tgen.org/home/research/research-divisions/cancer-and-cell-biology/targeted-treatments-for-tumors-(t3).aspx">T3</a>)].</p></div>
					</div>
					<div class="col-xs-12 col-md-3 eddy-sidebar">
						<ul class="eddy-sidebar-nav">
							<a href="#what-is-eddy" class="eddy-sidebar-nav-link"><li>What is EDDY?</li></a>
							<a href="#downloads" class="eddy-sidebar-nav-link"><li>Downloads</li></a>
							<a href="#system-requirements" class="eddy-sidebar-nav-link"><li>System Requirements</li></a>
							<a href="#file-formats" class="eddy-sidebar-nav-link"><li>File Formats</li></a>
							<a href="#operating-the-cli" class="eddy-sidebar-nav-link"><li>Operating the Command Line Interface</li></a>
							<a href="#post-processing-eddy-output" class="eddy-sidebar-nav-link"><li>Post-processing EDDY Output</li></a>
						</ul>
					</div>
				</div>
			</div>
		</div>
		<div class="row">
			<div class="col-xs-12 eddy-footer jumbotron ">
<p><a href="https://www.tgen.org/home/research/research-divisions/integrated-cancer-genomics.aspx">Integrated Cancer Genomics Division <span class="glyphicon glyphicon-link"></span></a></p>
				<p><a href="http://www.tgen.org"><img src="assets/imgs/TGen_Color_LOGO_medium.png"><span class="glyphicon glyphicon-link"></span></a></p>
			</div>
		</div>
	</div>
</body>
</html>
